subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvoker.java

```diff
@@ -154,8 +198,8 @@ public TransformerExecution(File primaryInput, Transformer transformer, File wor
             this.transformer = transformer;
             this.identityString = "transform/" + identityString;
             this.historyRepository = historyRepository;
-            this.outputDir = new File(workspace, "outputDirectory");
-            this.resultsFile = new File(workspace,  "results.bin");
+            this.outputDir = workspace;
+            this.resultsFile = new File(workspace.getParentFile(),  workspace.getName() + RESULTS_FILE_SUFFIX);
```

Comment:
This is violating scope, we shouldn't do this. Why not put the outputs in `$workspace/outputDirectory` and the results file in `$workspace/results.bin` or something?

Reply:
My main reason to do it like that is descibed in this commit comment: https://github.com/gradle/gradle/commit/0958b5244bd360309f95e3ea6b8ed813c9a86297

So if we change how we determine artifact ids, then we can use the other setup, where both files end up in the workspace. We could also use the `workspace` name instead of `outputDirectory`.

Another reason to use a sibling file is that we may run into long path issues on Windows. Although, I am not too worried about that now.

Reply:
I'm not saying we shouldn't put the workspace directory next to the result file. But the code shouldn't rely on implicit knowledge that "it's okay to put a file next to the workspace."

We could instead pass something like a `WorkspaceLocationProvider` that could give you either the output directory or the result file location, and pass that around.

Reply:
I introduced `TransformationWorkspace` and use that instead.